sycl/test/basic_tests/queue/queue_parallel_for_generic.cpp

```diff
@@ -0,0 +1,72 @@
+// UNSUPPORTED: cuda
+// CUDA does not support unnamed lambdas.
+//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsycl-unnamed-lambda %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==- queue_parallel_for_generic.cpp - SYCL queue parallel_for generic lambda -=//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <type_traits>
+
+int main() {
+  sycl::queue q{};
+  auto dev = q.get_device();
+  auto ctx = q.get_context();
+  constexpr int N = 8;
+
+  if (dev.get_info<sycl::info::device::usm_shared_allocations>()) {
+    auto A = static_cast<int *>(sycl::malloc_shared(N * sizeof(int), dev, ctx));
```

Comment:
```suggestion
  if (!dev.get_info<sycl::info::device::usm_shared_allocations>())
    return 0;
    
  auto A = static_cast<int *>(sycl::malloc_shared(N * sizeof(int), dev, ctx));
```
Please, use "early-exit" to improve readability.

Reply:
Done